File: leetcode/string/944. Delete Columns to Make Sorted/944. Delete Columns to Make Sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int columnsToDelete = 0;
        for(int i = 0; i < strs[0].size(); i++) {
            bool isNotLexicographically = false;
            for(int j = 0; j < strs.size() - 1; j++) {
                if(strs[j][i] > strs[j+1][i]) {
                    isNotLexicographically = true;
                    break;
                }
            }
            if (isNotLexicographically) columnsToDelete++;
        }
        return columnsToDelete;
    }
};

File: leetcode/string/944. Delete Columns to Make Sorted/944. Delete Columns to Make Sorted.rs
impl Solution {
    pub fn min_deletion_size(strs: Vec<String>) -> i32 {
        let mut columns_to_delete = 0;

        for i in 0..strs[0].len() {
            for j in 0..strs.len() - 1 {
                if strs[j].as_bytes()[i] > strs[j + 1].as_bytes()[i] {
                    columns_to_delete += 1;
                    break;
                }
            }
        }

        columns_to_delete
    }
}
